schemer/__init__.py

```diff
@@ -54,6 +54,10 @@ def apply_defaults(self, instance):
                 else:
                     instance[field] = default
 
+            elif isinstance(spec['type'], Schema) and 'populate_default_subdoc' in spec and spec['populate_default_subdoc']:
+                instance[field] = {}
+                field_type = spec['type']
+                field_type.apply_defaults(instance[field])
```

Comment:
reorder structure of this loop to allow for easier extensibility and to clean up method